### PR TITLE
Fix hot peer cache threshold unstable when interval unstable

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Before PR, we add peer to the hot peer cache when its interval is less than normal, which causes topn will increase a lot when the heartbeat interval is unstable. We calculate the threshold by the 60th item in the hot peer cache, so the threshold will reduce to the default threshold while a lot of peers are added to the hot peer cache.

### What is changed and how it works?

we add peer to the hot peer cache when its interval is less than normal only when it is hot too.

### Check List

Tests

- Unit test

### CODE REVIEW

**Review by rleungx — COMMENTED** [review]
LGTM, but the test is failed

**Review by nolouch — COMMENTED** [review]
lgtm, plz fix the test.

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
PTAL @rleungx @HunDunDM

File: server/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -385,11 +385,11 @@ func (f *hotPeerCache) updateHotPeerStat(newItem, oldItem *HotPeerStat, bytes, k
...
 		}
+		isHot := bytes/interval.Seconds() >= newItem.thresholds[byteDim] || keys/interval.Seconds() >= newItem.thresholds[keyDim]
+		if !isHot {
+			return nil
+		}
 		if interval.Seconds() >= RegionHeartBeatReportInterval {
-			isHot := bytes/interval.Seconds() >= newItem.thresholds[byteDim] || keys/interval.Seconds() >= newItem.thresholds[keyDim]
-			if !isHot {
-				return nil
-			}
```
Comment (Yisaer) [inline server/statistics/hot_peer_cache.go:392]: I'm wondering what's difference here? It seems the result won't be changed with the given `isHot` and `interval.Seconds() >= RegionHeartBeatReportInterval`

Comment (lhy1024) [reply server/statistics/hot_peer_cache.go:392]: it will directly return nil when the interval is less than RegionHeartBeatReportInterval

**Review by rleungx — APPROVED** [review]

**Pull request merged**